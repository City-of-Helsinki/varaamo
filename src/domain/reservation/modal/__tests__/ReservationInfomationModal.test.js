import React from 'react';
import toJSON from 'enzyme-to-json';

import { shallowWithIntl } from '../../../../../app/utils/testUtils';
import ReservationInformationModal from '../ReservationInformationModal';
import reservation from '../../../../common/data/fixtures/reservation';

const findButtonByLabel = (wrapper, label) => wrapper.find({ children: `ReservationInfoModal.${label}` });

describe('ReservationInformationModal', () => {
  const mockReservation = reservation.build({
    begin: '2019-08-14T14:00:00+03:00',
    end: '2019-08-14T15:00:00+03:00',
    state: 'confirmed',
  });

  const defaultProps = {
    reservation: mockReservation,
    onHide: jest.fn(),
    onSaveComment: jest.fn(),
    isOpen: true,
    onEditReservation: jest.fn(),
    t: path => path,
  };

  const getWrapper = props => shallowWithIntl(
    <ReservationInformationModal {...defaultProps} {...props} />,
  );

  test('renders correctly', () => {
    expect(toJSON(getWrapper())).toMatchSnapshot();
  });

  test('does not render deny and approve buttons by default', () => {
    const wrapper = getWrapper();

    expect(findButtonByLabel(wrapper, 'approveButton').length).toEqual(0);
    expect(findButtonByLabel(wrapper, 'denyButton').length).toEqual(0);
  });

  describe('when state is requested', () => {
    const getWrapperInRequestedState = props => getWrapper({
      reservation: { ...defaultProps.reservation, state: 'requested' },
      ...props,
    });

    test('render deny and approve buttons', () => {
      const wrapper = getWrapperInRequestedState();

      expect(findButtonByLabel(wrapper, 'approveButton').length).toEqual(1);
      expect(findButtonByLabel(wrapper, 'denyButton').length).toEqual(1);
    });
  });

  describe('refund policy', () => {
    test('rendered when resource is not null, when user is admin, when reservation is not a staff event and when the reservation has a price', () => {
      const getRefundPolicy = (wrapper) => {
        return wrapper.find({ id: 'refund-policy' });
      };
      const notAdmin = getWrapper({ isEditing: true, isAdmin: false });
      const admin = getWrapper({
        isEditing: true,
        isAdmin: true,
        reservation: {
          ...mockReservation,
          isStaffEvent: false,
          begin: new Date(2017, 10, 1, 9, 0, 0, 0).toJSON(),
          end: new Date(2017, 10, 1, 11, 0, 0, 0).toJSON(),
        },
        resource: {
          products: [
            {
              price: {
                type: 'per_period',
                period: '01:00',
                amount: 100,
              },
            },
          ],
        },
      });

      expect(getRefundPolicy(notAdmin).length).toEqual(0);
      expect(getRefundPolicy(admin).length).toEqual(1);
    });
  });
});
