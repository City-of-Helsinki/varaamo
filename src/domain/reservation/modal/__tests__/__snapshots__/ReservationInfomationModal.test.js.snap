// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservationInformationModal renders correctly 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  className="app-ReservationInformationModal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": [],
      "data": [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": [],
      "remove": [Function],
    }
  }
  onHide={[MockFunction]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      ReservationInfoModal.title
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <InjectT(ManageReservationsStatus)
      reservation={
        {
          "begin": "2019-08-14T14:00:00+03:00",
          "comments": "",
          "end": "2019-08-14T15:00:00+03:00",
          "event_description": "",
          "event_subject": "",
          "has_catering_order": false,
          "id": 1,
          "its_own": false,
          "resource": 1,
          "staff_event": false,
          "state": "confirmed",
          "url": "https://respa.koe.hel.ninja/v1/reservation/192388/",
          "user": {},
          "user_permissions": null,
        }
      }
    />
    <div
      className="app-ReservationInformationModal__information"
    >
      <div
        className="app-ReservationInformationModal__information__account"
      >
        <ReservationInformationModalContentRow
          content={
            <span>
              
            </span>
          }
          key="user_name"
          label={
            <span>
              common.userNameLabel
            </span>
          }
        />
        <ReservationInformationModalContentRow
          content={
            <span>
              
            </span>
          }
          key="user_email"
          label={
            <span>
              common.userEmailLabel
            </span>
          }
        />
      </div>
      <ReservationInformationModalContentRow
        content={
          <span>
            ke 14.08.2019 14:00 - 15:00
          </span>
        }
        key="reservation_time"
        label={
          <span>
            common.reservationTimeLabel
          </span>
        }
      />
      <ReservationInformationModalContentRow
        content={
          <span>
            
          </span>
        }
        key="resource"
        label={
          <span>
            common.resourceLabel
          </span>
        }
      />
      <InjectT(ReservationMetadata)
        customField={[Function]}
        reservation={
          {
            "begin": "2019-08-14T14:00:00+03:00",
            "comments": "",
            "end": "2019-08-14T15:00:00+03:00",
            "event_description": "",
            "event_subject": "",
            "has_catering_order": false,
            "id": 1,
            "its_own": false,
            "resource": 1,
            "staff_event": false,
            "state": "confirmed",
            "url": "https://respa.koe.hel.ninja/v1/reservation/192388/",
            "user": {},
            "user_permissions": null,
          }
        }
      />
    </div>
    <div
      className="app-ReservationInformationModal__edit-reservation-btn"
    >
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="primary"
        disabled={false}
        onClick={[Function]}
      >
        ReservationEditForm.startEdit
      </Button>
    </div>
    <div
      className="app-ReservationInformationModal__comments-section"
    >
      <ControlLabel
        bsClass="control-label"
        srOnly={false}
      >
        common.commentsLabel:
      </ControlLabel>
      <FormControl
        bsClass="form-control"
        className="app-ReservationInformationModal__comments-field"
        componentClass="textarea"
        onChange={[Function]}
        placeholder="common.commentsPlaceholder"
        rows={5}
        value=""
      />
      <div
        className="app-ReservationInformationModal__save-comment"
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="primary"
          disabled={false}
          onClick={[Function]}
        >
          ReservationInfoModal.saveComment
        </Button>
      </div>
    </div>
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="primary"
      disabled={false}
      onClick={[MockFunction]}
    >
      common.back
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
    >
      ReservationInfoModal.cancelButton
    </Button>
  </ModalFooter>
  <Connect(InjectT(UnconnectedReservationCancelModal))
    onEditReservation={[MockFunction]}
    parentToggle={[Function]}
    reservation={
      {
        "begin": "2019-08-14T14:00:00+03:00",
        "comments": "",
        "end": "2019-08-14T15:00:00+03:00",
        "event_description": "",
        "event_subject": "",
        "has_catering_order": false,
        "id": 1,
        "its_own": false,
        "resource": 1,
        "staff_event": false,
        "state": "confirmed",
        "url": "https://respa.koe.hel.ninja/v1/reservation/192388/",
        "user": {},
        "user_permissions": null,
      }
    }
    toggleShow={false}
  />
</Modal>
`;
